### PR TITLE
plugin emittance: remove cuSTL

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Replace cuSTL with PMacc natives.

# Tests

I compared the results of the first 800 steps for the Laser Wakefield example from this PR and the dev branch.
The results are in the first digits equal. The small difference I saw is coming from the random particle position in the used example.

### CODE REVIEW

File: include/picongpu/plugins/Emittance.hpp [hunk 1]
```diff
@@ -334,43 +331,29 @@ namespace picongpu
...
-                    inPlaneGPU[r_element] = 0;
-                    if(inPlaneGPU == pmacc::math::Int<simDim>::create(0))
-                        isGroupRoot = true;
-                }
-                auto* createReduce = new algorithm::mpi::Reduce<simDim>(zoneTransversalPlane, isGroupRoot);
-                if(isInGroup)
-                {
-                    planeReduce = createReduce;
-                    isPlaneReduceRoot = isGroupRoot;
+                    this->isPlaneReduceRoot = mpiReduce->hasResult(::pmacc::mpi::reduceMethods::Reduce{});
```
Comment (PrometheusPi) [inline include/picongpu/plugins/Emittance.hpp:343]: why is this now a member of the class and not just a local variable?

File: include/picongpu/plugins/Emittance.hpp [hunk 2]
```diff
@@ -334,43 +331,29 @@ namespace picongpu
...
-                    delete createReduce;
             }
 
-            /* Create communicator with ranks of each plane reduce root */
+            /* Create MPI communicator for openPMD IO with ranks of each plane reduce root */
             {
                 /* Array with root ranks of the planeReduce operations */
                 std::vector<int> planeReduceRootRanks(gc.getGlobalSize(), -1);
                 /* Am I one of the planeReduce root ranks? my global rank : -1 */
                 int myRootRank = gc.getGlobalRank() * isPlaneReduceRoot - (!isPlaneReduceRoot);
```
Comment (PrometheusPi) [inline include/picongpu/plugins/Emittance.hpp:353]: why is no `this->isPlaneReduceRoot` needed here?

**Review by PrometheusPi — APPROVED** [review]
looks formally good to me - just some questions

File: include/picongpu/plugins/Emittance.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/Emittance.hpp:343]: It was a member before. You can explicitly say via `this` it is a member or simply use the name. ONly if you have a local variable defined in the function with the same name and you like to assign to the class member you are required to use `this`

Comment (PrometheusPi) [reply include/picongpu/plugins/Emittance.hpp:343]: Thanks for the answer

File: include/picongpu/plugins/Emittance.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/plugins/Emittance.hpp:353]: In cases  where `isPlaneReduceRoot == false` we will have `X * 0 - 1 == -1` else `X * 1 - 0 == X`

Comment (PrometheusPi) [reply include/picongpu/plugins/Emittance.hpp:353]: answered by previous comment

**Pull request merged**